### PR TITLE
fix: STS AssumeRole returns empty credentials when endpoint has no path (#205)

### LINKED ISSUE #205: STS 获取临时凭证
minio::creds::AssumeRoleProvider provider;
minio::creds::Credentials credentials = provider.Fetch();

credentials.access_key
credentials.secret_key
credentials.session_token
打印为空

### PR DESCRIPTION
**Pull request by jiuker**
Fixes issue #205: temporary credentials (access_key / secret_key / session_token) returned by AssumeRoleProvider are empty. Root cause: when the STS endpoint has no path (e.g. http://host:9000), the SigV4 signature is computed with an empty CanonicalURI, while the actual request is sent to '/', causing signature verification to fail and leaving credentials empty.

Also aligns all STS provider request parameters with the minio-go reference implementation and fixes the XML root node paths used for credentials parsing.

Bug fix (issue #205):
- AssumeRoleProvider constructor: normalize an empty endpoint path to '/', keeping the signed CanonicalURI consistent with the actual request URI (conditional assignment only - does not override user-defined paths such as /minio/sts)
- Add an AssumeRoleProvider() regression test: success path asserts all three temporary credential fields are non-empty; failure path asserts the error is diagnosable via creds.err

Align STS provider parameters with minio-go:
- Add token_revoke_type to every provider (AssumeRoleProvider, WebIdentityClientGrantsProvider/ClientGrantsProvider/WebIdentityProvider, LdapIdentityProvider, CertificateIdentityProvider), added to the request body only when non-empty
- LdapIdentityProvider additionally gains policy, duration_seconds, and config_name
- Jwt struct gains access_token / refresh_token; the WebIdentity flow now sends WebIdentityAccessToken / WebIdentityRefreshToken
- All new parameters are appended at the end with defaults, backward compatible

XML parsing fix:
- All STS response parsing now uses the full root path with the Response prefix (e.g. AssumeRoleResponse/AssumeRoleResult), avoiding parse failures

Files changed: include/miniocpp/providers.h, src/providers.cc, tests/tests.cc

## Summary by CodeRabbit

* **New Features**
  * Added custom-token authentication for obtaining temporary credentials.
  * Added access and refresh token support for web identity authentication.
  * Added token-revocation settings across supported authentication providers.
  * Expanded LDAP and certificate authentication options, including policy, duration, and configuration settings.
  * Forwarded additional authentication parameters in temporary credential requests.

* **Bug Fixes**
  * Improved parsing of temporary credential responses.
  * Standardized default STS endpoint paths.
  * Added validation to detect incomplete credential responses.

### CODE REVIEW

File: tests/tests.cc [hunk 1]
```diff
@@ -1537,6 +1537,44 @@ class Tests {
...
+    if (!creds.err.String().empty()) {
+      // Failure path (issue #205): empty credential fields must be
+      // accompanied by a diagnosable error.
+      throw std::runtime_error(
+          "AssumeRoleProvider(): Fetch() failed with an error" +
+          creds.err.String());
+      return;
+    }
+
+    // Success path: temporary credentials must all be non-empty.
```
[the remark this answers is not in the document]
Comment (jiuker) [reply tests/tests.cc]: check with https://github.com/minio/minio-cpp/pull/247/commits/0f731f50a6f700f9c2e798a0ac758a071184ecda
